From cec579c33a7e00543b486076965aa27fe24b3c31 Mon Sep 17 00:00:00 2001
From: Matthew Toohey <contact@mtoohey.com>
Date: Thu, 16 Nov 2023 12:33:10 -0500
Subject: [PATCH] add kk_int8(un)box

---
 include/kklib/box.h | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/include/kklib/box.h b/include/kklib/box.h
index 58e5dedb..e0ed053c 100644
--- a/include/kklib/box.h
+++ b/include/kklib/box.h
@@ -245,6 +245,22 @@ static inline kk_box_t kk_int16_box(int16_t i, kk_context_t* ctx) {
 }
 #endif
 
+#if (KK_INTF_SIZE<=1)
+kk_decl_export int8_t  kk_int8_unbox(kk_box_t v, kk_context_t* ctx);
+kk_decl_export kk_box_t kk_int8_box(int8_t i, kk_context_t* ctx);
+#else
+static inline int8_t kk_int8_unbox(kk_box_t v, kk_context_t* ctx) {
+  kk_unused(ctx);
+  kk_intf_t i = kk_intf_unbox(v);
+  kk_assert_internal((i >= INT8_MIN && i <= INT8_MAX) || kk_box_is_any(v));
+  return (int8_t)(i);
+}
+static inline kk_box_t kk_int8_box(int8_t i, kk_context_t* ctx) {
+  kk_unused(ctx);
+  return kk_intf_box(i);
+}
+#endif
+
 #if (KK_INTF_SIZE == 8) && KK_BOX_DOUBLE64
 kk_decl_export kk_box_t kk_double_box(double d, kk_context_t* ctx);
 kk_decl_export double   kk_double_unbox(kk_box_t b, kk_borrow_t borrow, kk_context_t* ctx);
-- 
2.42.0

